COLLABORATOR_GUIDE.md

```diff
@@ -461,7 +462,20 @@ It is important to mention that there are some rules on our Vercel Deployments s
   - Hence if Builds fail unexpectedly, make sure that your dependency that is being used during build-time is on `dependencies` and not `devDependencies`. Checkout out [DEPENDENCY_PINNING.md](./DEPENDENCY_PINNING.md) for more information.
 - Our sponsorship with Vercel is maintained by the OpenJS Foundation
 
-### Seeking additional clarification
+### Why we have a `.vscode` folder
+
+The repository defines an optimized configuration for code editing. This is optional and is not required to contribute to the project. However, the settings and extensions specified help create a uniform and more efficient developer experience. This configuration is found in the `.vscode` directory:
+
+- `extensions.json` suggests VSCode extensions that make the editor more compatible with the code. For example, a Tailwind extensions creates auto-complete intellisense for tailwind styles within our components. Eslint, prettier, and editorconfig extensions read their respective config files and automatically format or lint code as it is written. This helps save CI feedback loops when a contribution does not meet our standards.
+- `settings.json` contains some common sense defaults that aide development and enforce consistency across the codebase. For example, we want files formatted on save and we want prettier to be used as the formatter. Without these settings, new contributors may have different authoring experiences when contributing, leading to inconsistent code and CI failures.
```

Comment:
Worth mentioning that one of the important settings is that we disable VSCode defaults `CSS parser` and use Stylelint since we use custom CSS syntax, and if we didn't do that, the IDE would be complaining about things that are completely OK.